src/lights/LightShadow.js

```diff
@@ -4,57 +4,59 @@ import { Vector3 } from '../math/Vector3.js';
 import { Vector4 } from '../math/Vector4.js';
 import { Frustum } from '../math/Frustum.js';
 
-function LightShadow( camera ) {
+class LightShadow {
 
-	this.camera = camera;
+	constructor( camera ) {
 
-	this.bias = 0;
-	this.normalBias = 0;
-	this.radius = 1;
+		Object.defineProperty( this, 'isLightShadow', { value: true } );
 
-	this.mapSize = new Vector2( 512, 512 );
+		this.camera = camera;
 
-	this.map = null;
-	this.mapPass = null;
-	this.matrix = new Matrix4();
+		this.bias = 0;
+		this.normalBias = 0;
+		this.radius = 1;
 
-	this.autoUpdate = true;
-	this.needsUpdate = false;
+		this.mapSize = new Vector2( 512, 512 );
 
-	this._frustum = new Frustum();
-	this._frameExtents = new Vector2( 1, 1 );
+		this.map = null;
+		this.mapPass = null;
+		this.matrix = new Matrix4();
 
-	this._viewportCount = 1;
+		this.autoUpdate = true;
+		this.needsUpdate = false;
 
-	this._viewports = [
+		this._frustum = new Frustum();
+		this._frameExtents = new Vector2( 1, 1 );
 
-		new Vector4( 0, 0, 1, 1 )
+		this._viewportCount = 1;
 
-	];
+		this._viewports = [
 
-}
+			new Vector4( 0, 0, 1, 1 )
+
+		];
 
-Object.assign( LightShadow.prototype, {
+		this._projScreenMatrix = new Matrix4();
 
-	_projScreenMatrix: new Matrix4(),
+		this._lightPositionWorld = new Vector3();
 
-	_lightPositionWorld: new Vector3(),
+		this._lookTarget = new Vector3();
 
-	_lookTarget: new Vector3(),
+	}
```

Comment:
Which should be updated? is it these ones?

Reply:
Hmm, this is hard to work with / review. 

Do you mind splitting this PR in multiple ones? (One per file)

Reply:
I think if we would have made single PR per file from the start we would have been done with this transition.
Right now a single file is blocking many others.

Reply:
sure